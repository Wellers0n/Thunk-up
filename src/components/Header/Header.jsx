import React from "react";
import AppBar from "@material-ui/core/AppBar";
import Toolbar from "@material-ui/core/Toolbar";
import Typography from "@material-ui/core/Typography";
import Button from "@material-ui/core/Button";
import IconButton from "@material-ui/core/IconButton";
import MenuIcon from "@material-ui/icons/Menu";
import useStyles from "./Style";
import { connect } from "react-redux";
import FetchList from './../../actions/FetchList'
import { bindActionCreators } from "redux";


function ButtonAppBar(props) {
  const {FetchList, data, error, loading} = props
  const classes = useStyles();
  return (
    <div className={classes.root}>
      <AppBar position="static">
        {console.log(data, error, loading)}
        <Toolbar>
          <IconButton
            edge="start"
            className={classes.menuButton}
            color="inherit"
            aria-label="menu"
          >
            <MenuIcon />
          </IconButton>
          <Typography
            data-testid="title-header"
            variant="h6"
            className={classes.title}
          >
            Home
          </Typography>
          <Button onClick={FetchList} color="inherit" data-testid="button-search">
            Search list
          </Button>
        </Toolbar>
      </AppBar>
    </div>
  );
}

const mapStateToProps = store => ({
  data: store.FetchList.data,
  error: store.FetchList.error,
  loading: store.FetchList.loading

});

const mapDispatchToProps = dispatch => ({
  FetchList: bindActionCreators(FetchList, dispatch)
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(ButtonAppBar);
