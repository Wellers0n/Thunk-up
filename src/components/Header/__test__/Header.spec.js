import React from "react";
import { render, waitForElement, fireEvent } from "@testing-library/react";
import Header from "../Header";

describe("App tests Header", () => {
    it("should render header with name 'search list' ", async () => {
        const {getByTestId, getByText} = render(<Header/>)

        const button = await waitForElement(() => getByTestId("button-search"))
        expect(button.textContent).toBe("Search list")
    })

    it("should render header with name 'Home' ", async () => {
        const {getByTestId, getByText} = render(<Header/>)

        const button = await waitForElement(() => getByTestId("title-header"))
        expect(button.textContent).toBe("Home")
    })

    it("should render a snapshot header ", async () => {
        const {container} = render(<Header/>)

        expect(container.firstChild).toMatchSnapshot();
    })
})

