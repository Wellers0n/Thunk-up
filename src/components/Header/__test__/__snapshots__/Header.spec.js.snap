// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App tests Header should render a snapshot header  1`] = `
<div
  class="makeStyles-root-243"
>
  <header
    class="MuiPaper-root MuiPaper-elevation4 MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary"
  >
    <div
      class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
    >
      <button
        aria-label="menu"
        class="MuiButtonBase-root MuiIconButton-root makeStyles-menuButton-244 MuiIconButton-colorInherit MuiIconButton-edgeStart"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
      <h6
        class="MuiTypography-root makeStyles-title-245 MuiTypography-h6"
        data-testid="title-header"
      >
        Home
      </h6>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-colorInherit"
        data-testid="button-search"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          Search list
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </header>
</div>
`;
